main.c

```diff
@@ -1295,8 +1296,22 @@ int main(int argc, char **argv) {
 		create_surface(surface);
 	}
 
+	if (state.ext_session_lock_manager_v1) {
+		while (!state.locked) {
+			if (wl_display_dispatch(state.display) < 0) {
+				swaylock_log(LOG_ERROR, "wl_display_dispatch() failed");
+				return 2;
+			}
+		}
+	} else {
+		if (wl_display_roundtrip(state.display) < 0) {
+			swaylock_log(LOG_ERROR, "wl_display_roundtrip() failed");
+			return 2;
+		}
+		state.locked = true;
```

Comment:
If we set this in the layer shell configure handler, we save branching here and get a better indication of locking than a single blind roundtrip.

On the other hand, this code will be deleted almost immediately so I don't care too much about it... :)

Reply:
I'd rather keep the changes for the legacy code-path minimal.